zipline/pipeline/engine.py

```diff
@@ -273,6 +276,12 @@ def _inputs_for_term(self, term, workspace, graph):
             out.append(input_data)
         return out
 
+    def get_loader(self, term):
+        if term is AssetExists():
```

Comment:
I becomes one of the keys in the `atomic_groups`, but that key is never looked up later because that term is already in the workspace.